web/src/components/Storyblok/StoryblokBlogPost/StoryblokBlogPost.tsx

```diff
@@ -0,0 +1,122 @@
+import { StoryblokComponent } from "@storyblok/react";
+import React, { useEffect, useRef } from "react";
+import { StoryblokStory } from "storyblok-generate-ts";
+import { StoryblokRichtext } from "storyblok-rich-text-react-renderer";
+
+import { Breadcrumbs, Breadcrumb } from "@nice-digital/nds-breadcrumbs";
+import { Grid, GridItem } from "@nice-digital/nds-grid";
+import { PageHeader } from "@nice-digital/nds-page-header";
+
+import { NewsLetterSignup } from "@/components/NewsLetterSignUp/NewsLetterSignup";
+import { type Breadcrumb as TypeBreadcrumb } from "@/types/Breadcrumb";
+import { AuthorStoryblok, BlogPostStoryblok } from "@/types/storyblok";
+
+import { NewsPageHeaderFooter } from "../NewsPageHeader/NewsPageHeaderFooter/NewsPageHeaderFooter";
+import { AuthorList } from "../StoryblokAuthor/AuthorList/AuthorList";
+import { StoryblokImage } from "../StoryblokImage/StoryblokImage";
+import { StoryblokRichText } from "../StoryblokRichText/StoryblokRichText";
+
+import styles from "./StoryblokBlogPost.module.scss";
+
+// overrride the default type to include the content to use the richtext renderer type from package
+export interface RichTextBlogPostStoryblok extends BlogPostStoryblok {
+	content: StoryblokRichtext;
+}
+
+export interface StoryblokBlogPostProps {
+	blok: RichTextBlogPostStoryblok;
+	breadcrumbs?: TypeBreadcrumb[];
+}
+
+export const StoryblokBlogPost = ({
+	blok,
+	breadcrumbs,
+}: StoryblokBlogPostProps): React.ReactElement => {
+	const imageRef = useRef<HTMLImageElement>(null);
+	const articleRef = useRef<HTMLDivElement>(null);
+
+	useEffect(() => {
+		const handleResize = () => {
+			// set the offset for the featured image
+			if (articleRef.current && imageRef.current) {
+				articleRef.current.style.setProperty(
+					"--featuredImageOffset",
+					`${Math.floor(imageRef.current.height / 1.75)}px`
+				);
+			}
+		};
+
+		window.addEventListener("resize", handleResize);
```

Comment:
Would you mind debouncing or throttling this, just so we can keep performance in check on rubbish systems/browsers?